src/kem/kyber/kem_kyber_512.c

```diff
@@ -46,7 +46,35 @@ extern int PQCLEAN_KYBER512_AARCH64_crypto_kem_enc(uint8_t *ct, uint8_t *ss, con
 extern int PQCLEAN_KYBER512_AARCH64_crypto_kem_dec(uint8_t *ss, const uint8_t *ct, const uint8_t *sk);
 #endif
 
+#if defined(OQS_ENABLE_LIBJADE_KEM_kyber_512)
+extern int libjade_kyber512_ref_keypair(uint8_t *pk, uint8_t *sk);
+extern int libjade_kyber512_ref_enc(uint8_t *ct, uint8_t *ss, const uint8_t *pk);
+extern int libjade_kyber512_ref_dec(uint8_t *ss, const uint8_t *ct, const uint8_t *sk);
+#endif
+
+#if defined(OQS_ENABLE_LIBJADE_KEM_kyber_512_avx2)
+extern int libjade_kyber512_avx2_keypair(uint8_t *pk, uint8_t *sk);
+extern int libjade_kyber512_avx2_enc(uint8_t *ct, uint8_t *ss, const uint8_t *pk);
+extern int libjade_kyber512_avx2_dec(uint8_t *ss, const uint8_t *ct, const uint8_t *sk);
+#endif
+
+
 OQS_API OQS_STATUS OQS_KEM_kyber_512_keypair(uint8_t *public_key, uint8_t *secret_key) {
+#if defined(OQS_LIBJADE_BUILD) && (defined(OQS_ENABLE_LIBJADE_KEM_kyber_512))
+#if defined(OQS_ENABLE_LIBJADE_KEM_kyber_512_avx2)
+#if defined(OQS_DIST_BUILD)
+	if (OQS_CPU_has_extension(OQS_CPU_EXT_AVX2) && OQS_CPU_has_extension(OQS_CPU_EXT_BMI2) && OQS_CPU_has_extension(OQS_CPU_EXT_POPCNT)) {
+#endif /* OQS_DIST_BUILD */
```

Comment:
Wow - one impressive set of config options... Do we have enough new CI that triggers (and not) this? I found mostly MINIMAL_BUILD configs in GH actions files. Are these sufficient?

Reply:
I'm not sure I understand your question entirely but if you're asking "Do the AVX2 implementations from libjade get tested in CI?" then yes they do. These config options are standard for all AVX2 implementations even for those not from libjade.

Reply:
@praveksharma Sorry for not being clearer before. What I meant to ask is whether there is CI that tests all config options possible, thus triggering all code bases embedded/enabled by the various defines/config options that may be chosen by users, i.e., libjade ref and avx2 code, pqcrystals ref and avx2 and pqclean aarch64 code?

Reply:
@baentsch Thank you for clarifying! I believe the aarch64 code does end up getting tested on our ARM tests but you're right our CI tests don't explicitly use the OQS_USE_CPUFEATURE_INSTRUCTIONS option. So it would seem we do not actually test all options available to users in CONFIGURE.md. I can create a separate issue to track this.